Resources/public/js/models/ez-locationmodel.js

```diff
@@ -42,6 +42,27 @@ YUI.add('ez-locationmodel', function (Y) {
             } else {
                 callback("Only read operation is supported at the moment");
             }
+        },
+
+        /**
+         * Moves location to trash. Callback is directly passed to ContentService.moveSubtree location
+         *
+         * @method trash
+         * @param {Object} options the options for moving to trash
+         * @param {Object} options.api (required) the JS REST client instance
+         * @param {Function} callback
+         */
+        trash: function (options, callback) {
+            var api = options.api,
+                location = this;
+
+            api.getContentService().loadRoot(function (error, response) {
+                var trashPath = response.document.Root.trash._href;
```

Comment:
you should first check the error before using the response, for what ever reason, the loadRoot call might fail